File: SupplierPortal.Web/Modules/ServerTypes/Market/OfferDetailForm.ts
import { IntegerEditor, LookupEditor, StringEditor, DecimalEditor, EnumEditor, DateEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { Currency } from "./Enum.Currency";

export interface OfferDetailForm {
    OfferId: IntegerEditor;
    ItemId: LookupEditor;
    Kdv: StringEditor;
    Unit: StringEditor;
    Quantity: DecimalEditor;
    Price: DecimalEditor;
    TotalPrice: DecimalEditor;
    Curency: EnumEditor;
    Description: StringEditor;
    Date: DateEditor;
}

export class OfferDetailForm extends PrefixedContext {
    static readonly formKey = 'Market.OfferDetail';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!OfferDetailForm.init)  {
            OfferDetailForm.init = true;

            var w0 = IntegerEditor;
            var w1 = LookupEditor;
            var w2 = StringEditor;
            var w3 = DecimalEditor;
            var w4 = EnumEditor;
            var w5 = DateEditor;

            initFormType(OfferDetailForm, [
                'OfferId', w0,
                'ItemId', w1,
                'Kdv', w2,
                'Unit', w2,
                'Quantity', w3,
                'Price', w3,
                'TotalPrice', w3,
                'Curency', w4,
                'Description', w2,
                'Date', w5
            ]);
        }
    }
}

[Currency]; // referenced types

File: SupplierPortal.Web/Modules/Market/OfferDetail/OfferDetailForm.cs
using Serenity.ComponentModel;
using SupplierPortal.Inventory.Enum;
using SupplierPortal.Market.Enum;
using System;

namespace SupplierPortal.Market.Forms;

[FormScript("Market.OfferDetail")]
[BasedOnRow(typeof(OfferDetailRow), CheckNames = true)]
public class OfferDetailForm
{
    [Tab("Offer Detail"), IgnoreName]
    [Visible(false)]
    public int OfferId { get; set; }
    public int ItemId { get; set; }
    public string Kdv { get; set; }
    public string Unit { get; set; }
    public decimal Quantity { get; set; }
    public decimal Price { get; set; }
    public decimal TotalPrice { get; set; }
    public Currency Curency { get; set; }
    public string Description { get; set; }
    public DateTime Date { get; set; }


}